pkg/version/version.go

```diff
@@ -23,3 +32,87 @@ func MaybePrintVersionAndExit(printVersion bool) {
 	}
 	log.Infof("running conduit version %s", Version)
 }
+
+type versionStatusChecker struct {
+	version string
+	client  pb.ApiClient
+}
+
+func (v versionStatusChecker) SelfCheck() []*healthcheckPb.CheckResult {
+	cliVersion := v.version
+	cliIsUpToDate := &healthcheckPb.CheckResult{
+		Status:           healthcheckPb.CheckStatus_OK,
+		SubsystemName:    "conduit-version",
+		CheckDescription: "CLI is up-to-date",
+	}
+
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		cliIsUpToDate.Status = healthcheckPb.CheckStatus_ERROR
+		cliIsUpToDate.FriendlyMessageToUser = err.Error()
+		return []*healthcheckPb.CheckResult{cliIsUpToDate}
+	}
+	if cliVersion != latestVersion {
+		cliIsUpToDate.Status = healthcheckPb.CheckStatus_FAIL
+	}
+
+	controlPlaneIsUpToDate := &healthcheckPb.CheckResult{
+		Status:           healthcheckPb.CheckStatus_OK,
+		SubsystemName:    "conduit-version",
+		CheckDescription: "control plane is up-to-date",
+	}
+
+	controlPlaneVersion, err := getServerVersion(v.client)
+	if err != nil {
+		controlPlaneIsUpToDate.Status = healthcheckPb.CheckStatus_ERROR
+		controlPlaneIsUpToDate.FriendlyMessageToUser = err.Error()
+		return []*healthcheckPb.CheckResult{controlPlaneIsUpToDate}
+	}
+	if controlPlaneVersion != latestVersion {
+		controlPlaneIsUpToDate.Status = healthcheckPb.CheckStatus_FAIL
```

Comment:
Right now, when I run this branch on my box I get this:
```shell
pcalcado@arbeitsziege.local$ curl https://versioncheck.conduit.io/version.json
{"version": "v0.3.0"}
pcalcado@arbeitsziege.local$  bin/conduit check
kubernetes-api: can initialize the client.......................................[ok]
kubernetes-api: can query the Kubernetes API....................................[ok]
kubernetes-api: is running the minimum Kubernetes API version...................[ok]
conduit-api: can query the Conduit API..........................................[ok]
conduit-api[telemetry]: control plane can use telemetry service.................[ok]
conduit-version: CLI is up-to-date..............................................[FAIL]  --
conduit-version: control plane is up-to-date....................................[FAIL]  --

Status check results are [FAIL]
```

Should we set a `controlPlaneIsUpToDate.FriendlyMessageToUser ` to something like `"we are running version x but the latest version is y"` so that it is displayed here instead of the `--`?